lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py

```diff
@@ -3,61 +3,130 @@
 
 Sample usage::
 
-    python lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py --app-csv "edx-platform Apps Ownership.csv"
+    python lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py --repo-csv "Individual Repo Ownership.csv" --app-csv "edx-platform Apps Ownership.csv"
 
-Sample CSV input::
+Or for more details::
 
-    Path,owner.squad
-    ./common/djangoapps/xblock_django,team-red
-    ./openedx/core/djangoapps/xblock,team-red
-    ./lms/djangoapps/badges,team-blue
+    python lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py --help
 
-Sample output::
-
-    # Copy results into appropriate config yml file.
-    CODE_OWNER_MAPPINGS:
-      team-blue:
-      - badges
-      team-red:
-      - xblock_django
-      - openedx.core.djangoapps.xblock
 
 """
-import os
-import re
 import csv
 import click
+import logging
+import os
+import re
+
+log = logging.getLogger(__name__)
+
+# TODO: Clean this up.
+# - Move to spreadsheet or leave in code?
+# - If in code, reverse mapping so enterprise will be grouped.
+# - Use repo-name vs repo-url to make cleaner.
+# See clean_unmapped_view_modules.py for updating
+EXTERNAL_APPS = {
```

Comment:
1. We simply need to store a mapping of apps (seen in NewRelic), and what repo they live in.
2. For edx owned repos, we already know the owners.  The mapping of repo to app list could live either in the spreadsheet or this script.  For now, it is in the script.  Ultimately, either way, it gets translated to remote configuration.
3. For non-edx owned repos, we don't yet have ownership mapped.  This either needs to be done in the spreadsheet, or these apps can be mapped to a repo or folder that already has ownership assigned. These cases are the tuples you'll find below.

Reply:
From my discussion with @nasthagiri:
* Mapping from app to repo will stay in code.
* Third party dependencies ownership will move to spreadsheet.
* Mapping data structure will be cleaned up and separated between third-party and edx-owned.